.github/actions/publish/action.yml

```diff
@@ -0,0 +1,30 @@
+name: Publish Artifacts
+description: 'Publish artifacts to Github Release'
+inputs:
+  token:
+    description: 'Token to use for publishing.'
+    required: true
+  tag:
+    description: 'Tag to upload artifacts to.'
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - name: Git clean
+      shell: bash
+      run: git clean -f
```

Comment:
What is the reason for this clean?  Feels like cleans in actions could lead to unexpected results when combined into a workflow.  Or make them difficult to move as more workflows/actions get added over time.

I use `git clean -xdf` locally sometimes, consider the x and d arguments?

Reply:
The clean is because `goreleaser`  will fail if git is in a dirty state. Since the publish step is run after unit tests, there's usually a coverage file or similar created. 

I think I agree with what you're getting at though - this is a brute force way of doing it; the publish action is acting defensively. It might be better to have the unit test step ensure git is clean instead.